### DISCUSSION TITLE
Client generator does not generate projections for "recursive" schemas?

### DISCUSSION THREAD

**Discussion by ptorkko** [Netflix/dgs-framework#1416] [Q&A] [answered]
Given the partial schema of

```
type Hierarchy {
  code: String!
  children(...): HierarchyConnection
}
type HierarchyConnection {
  edges: [HierarchyEdge]
  pageInfo: PageInfo
}
type HierarchyEdge {
  node: Hierarchy
  cursor: String
}
...
```

version `5.6.9` of the `com.netflix.dgs.codegen` gradle plugin seems to only generate projection of

```
HierarchyProjectionRoot()
  .code()
  .children()
    .edges()
      .node()
        .code()
        .children() // this is omitted
```
Regardless of the `maxProjectionDepth` property of the `generateJava` task, it seems the recursion always stops at the first loop reference in the schema.

Is there any way around this?

**Comment by srinivasankavitha** [comment, marked as answer]
With the current implementation of code generation, it  will stop once it
detects a cycle and we don't have a configuration option to work around it.
The maxProjectionDepth is applicable when there are no cycles.

We are in the process of reworking the API and the implementation to fix a
bunch of issues with the current implementation. We will likely release
something end of this month. IN the meantime, I would recommend just using
query strings if this is a blocking issue for you.

**Reply by Vano2776** [reply]
Already have kotlin dsl projections (can use with flag generateKotlinClosureProjections = true), why this not have in the documentation?

**Reply by andresmonc** [reply]
@srinivasankavitha hello - are we still planning to have a fix for this out today?

**Comment by srinivasankavitha** [comment]
Yes, today or tomorrow latest. This will be released as an experimental
feature since it has a slightly new API

**Reply by andresmonc** [reply]
thanks for the quick response - will lookout for that release

**Comment by srinivasankavitha** [comment]
The release is out:
https://github.com/Netflix/dgs-codegen/releases/tag/v5.7.0

You will need a new config `generateClientv2` to use the newer
implementation. Also note that the API has changed very slightly, but it is
also experimental, so we may change it to make improvements in the future.

**Reply by andresmonc** [reply]
Working great!!!
